### ISSUE TITLE
Compiler version comparison logic may not be valid

### ISSUE THREAD

**Issue by BornThisWay** [intel/linux-sgx#887]
https://github.com/intel/linux-sgx/blob/4887b30b519d7bfc9f5de6e6330bc0034cce3d75/SampleCode/LocalAttestation/EnclaveInitiator/Makefile#L82

On my machine, shouldn't the result be 0?
```
[root@localhost linux-sgx-sgx_2.15.1]# expr "`cc -dumpversion`" \< "4.9"
1
[root@localhost linux-sgx-sgx_2.15.1]# cc -dumpversion
10.3.1
```

**Comment by lzha101** [comment]
Yes. This version check should be updated, similar as the top buildenv.mk
https://github.com/intel/linux-sgx/blob/4887b30b519d7bfc9f5de6e6330bc0034cce3d75/buildenv.mk#L90-L94
